tools/power_analysis/wpilog.py

```
@@ -0,0 +1,302 @@
+    timestamp_len = ((header >> 4) & 0x7) + 1
+
+    entry_id = int.from_bytes(data[pos : pos + id_len], "little")
+    pos += id_len
+    payload_len = int.from_bytes(data[pos : pos + payload_len_len], "little")
+    pos += payload_len_len
+    timestamp_us = int.from_bytes(data[pos : pos + timestamp_len], "little")
+    pos += timestamp_len
+
+    payload = data[pos : pos + payload_len]
+    if len(payload) < payload_len:
+        raise IndexError("truncated record payload")
```

Ruff (0.16.3)

[warning] Avoid specifying long messages outside the exception class

(TRY003)